actioncable/app/assets/javascripts/action_cable/connection.coffee

```diff
@@ -46,12 +51,21 @@ class ActionCable.Connection
     else
       @open()
 
+  getProtocol: ->
+    @webSocket?.protocol
+
+  isProtocolSupported: ->
+    @getProtocol() in protocols and @getProtocol() isnt unsupported_protocol
```

Comment:
My problem is just that I don't see why we need to do insert the extra protocol at runtime. My understanding is that we can do all that upfront. Maybe that's an insufficient understanding of the domain.

I'll try rereading your comments tomorrow, see if my brain is more receptive then :smile:

Thanks for all your effort on this! This is going to be so cool to have in :heart: